lib/karafka/cli/topics.rb

```diff
@@ -10,26 +10,51 @@ class Topics < Base
       )
 
       desc 'Allows for the topics management'
+
+      option(
+        :detailed_exitcode,
+        'Exists with 0 when no changes, 1 when error and 2 when changes present or applied',
+        TrueClass,
+        %w[
+          --detailed_exitcode
+        ]
+      )
+
+      # We exit with 0 if no changes happened
+      NO_CHANGES_EXIT_CODE = 0
+
+      # When any changes happened (or could happen) we return 2 because 1 is default when Ruby
+      # crashes
+      CHANGES_EXIT_CODE = 2
+
+      private_constant :NO_CHANGES_EXIT_CODE, :CHANGES_EXIT_CODE
+
       # @param action [String] action we want to take
       def call(action = 'missing')
-        case action
-        when 'create'
-          Topics::Create.new.call
-        when 'delete'
-          Topics::Delete.new.call
-        when 'reset'
-          Topics::Reset.new.call
-        when 'repartition'
-          Topics::Repartition.new.call
-        when 'migrate'
-          Topics::Migrate.new.call
-        when 'align'
-          Topics::Align.new.call
-        when 'plan'
-          Topics::Plan.new.call
-        else
-          raise ::ArgumentError, "Invalid topics action: #{action}"
-        end
+        detailed_exit_code = options.fetch(:detailed_exitcode, false)
+
+        changes = case action
```

Comment:
Can we extract this `case` block to a separate method?

Reply:
I think it's clear in one method and does not violate rubocop rules. I would only extract the execution because its duplicated.